src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs

```diff
@@ -1326,12 +1296,28 @@ internal bool CloseConnection
                 return (Status.ErrorCode == SecurityStatusPalErrorCode.ContextExpired);
             }
         }
+        internal void SetPayload(ReadOnlySpan<byte> payload)
+        {
+            Debug.Assert(Payload == null);
+            Size = payload.Length;
+
+            if (Size > 0)
+            {
+                Payload = ArrayPool<byte>.Shared.Rent(Size);
+                payload.CopyTo(new Span<byte>(Payload, 0, Size));
+            }
+        }
 
-        internal ProtocolToken(byte[]? data, SecurityStatusPal status)
+        internal ReadOnlyMemory<byte> AsMemory() => new ReadOnlyMemory<byte>(Payload, 0, Size);
+
+        internal void Clear()
```

Comment:
Not fan of the name here, would prefer a name that implies that we are returning the pooled buffer and not just zero-ing the held memory. But I can't think of any right now.

Reply:
suggestion?

Reply:
`ReleasePayload` / `ReturnPayloadBuffer` ?

Reply:
I like `ReleasePayload`, updated.